misk-feature-testing/src/main/kotlin/misk/feature/testing/FakeFeatureFlags.kt

```diff
@@ -87,6 +102,14 @@ class FakeFeatureFlags @Inject constructor() : AbstractIdleService(),
     overrides[MapKey(feature)] = value
   }
 
+  fun <T> override(feature: Feature, value: T, clazz: Class<T>) {
+    overrides[MapKey(feature)] = moshi.get().adapter(clazz).toJson(value)
```

Comment:
there’s a dangerous feature that might be useful here...
```
 moshi.get().adapter(clazz).failOnUnknown().toJson(value)
```
The upside is it’ll catch typos where you say `shardCount` but wanted `shard_count`. The downside is if somebody does a deploy that takes away a field it’ll crash here rather than silently failing.

Reply:
... actually maybe we can have our cake and eat it too.

```
  val decoded = try { 
      moshi.get().adapter(clazz).failOnUnknown().toJson(value)
  } catch (e: JsonDataException) {
      logger.error(e)
      moshi.get().adapter(clazz).toJson(value)
  }
```


```

Reply:
nice. I like that. I initial avoided `failOnUnknown` because it makes rollouts of changes difficult.